cmd/mcp/README.md

```diff
@@ -0,0 +1,278 @@
+  - `namespace` - The namespace of the resource (required for namespaced resources).
+
+### Multi-cluster tools
+
+The MCP server provides a set of tools for multi-cluster operations:
+
+- `get_kubeconfig_contexts`: This tool retrieves the Kubernetes clusters contexts found in the kubeconfig.
+  - No arguments required
+- `set_kubeconfig_context`: This tool sets the context to a specific cluster for the current session.
+  - `name` - The name of the context to set (required).
+
+Note that the `set_kubeconfig_context` tool does not alter the kubeconfig file,
+it only sets the context for the current session. 
+
+### Documentation tool
+
+The MCP server provides a tool for searching the Flux documentation:
+
+- `search_flux_docs`: This tool searches the latest Flux documentation and returns up-to-date information.
+  - `query` - The search query (required).
+  - `limit` - The maximum number of results to return (default 1).
+
+Note that most AI models are trained on data up to a certain date and may not have the latest information.
+When asking questions about the Flux APIs or features, it is recommended to append the following sentence
+to the prompt: `Answer using the Flux docs search tool.`.
+
+## Security Considerations
+
+The data returned by the MCP server may contain sensitive information found in Flux resources,
+such as container images, Git repository URLs, and Helm chart names.
+By default, the MCP Server masks the values of the Kubernetes Secrets data values,
+but it is possible to disable this feature by setting the `--mask-secrets=false` flag.
+
+The MCP server exposes tools that alter the state of the cluster, such as suspending,
+resuming, and deleting Flux resources. To disable these tools, the MCP server can be
+configured to run in read-only mode by setting the `--read-only` flag.
+
+The MCP server uses the `KUBECONFIG` environment variable to read the configuration and
+authenticate to Kubernetes clusters. It is possible to specify a user or service account
+that the MCP server will impersonate when connecting to the cluster.
+
+Example configuration for impersonating a service account with read-only permissions:
+
+```json
+{
+  "mcpServers": {
+    "flux-operator-mcp": {
+      "command": "/Users/stefanprodan/src/flux-operator/bin/flux-operator-mcp",
+      "args": [
+        "serve",
+        "--read-only",
+        "--kube-as=system:serviceaccount:flux-system:flux-operator"
+      ],
+      "env": {
+        "KUBECONFIG": "/Users/stefanprodan/.kube/config"
+      }
+    }
+  }
+}
+```
+
+On the local machine file system, the MCP server performs only read operations.
+It reads the `KUBECONFIG` environment variable then reads the config file.
+If the config file contains `exec` instructions to authenticate to the cluster,
+the MCP server will execute the command in the same way as `kubectl` does.
+Note that running the MCP server in a container is possible
+only if the config contains static credentials.
```

Comment:
```suggestion
only if the container file system has the binaries used in the config.
```

Reply:
The binaries will not work, you need SSO with a browser prompt, access to various other config files in the home dir, etc.